cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/BullseyeParser.java

```diff
@@ -200,46 +195,96 @@ private void saveConditions(CoverageMeasures fileMeasuresBuilderIn) {
     totalcoveredconditions = 0;
   }
 
-  private void updateMeasures(String kind, String event, String line, CoverageMeasures fileMeasuresBuilderIn) {
+  private static void updateMeasures(String kind, String event, String line, CoverageMeasures fileMeasuresBuilderIn) {
 
-    if ("decision".equalsIgnoreCase(kind) || "condition".equalsIgnoreCase(kind)) {
-      if ("condition".equalsIgnoreCase(kind)) {
+    switch (kind.toLowerCase(Locale.ENGLISH)) {
+      case "catch":
+      case "decision":
+      case "for-range-body":
+      case "switch-label":
+      case "try":
+        totaldecisions++;
+        setTotalCoveredDecisions(event);
+        break;
+      case "condition":
         totalconditions += 2;
-        totalcoveredconditions += 1;
+        setTotalCoveredConditions(event);
+        break;
+      case "function":
         if ("full".equalsIgnoreCase(event)) {
-          totalcoveredconditions += 1;
-        }
-        if ("none".equalsIgnoreCase(event)) {
-          totalcoveredconditions -= 1;
+          fileMeasuresBuilderIn.setHits(Integer.parseInt(line), 1);
         }
-      } else {
-        totaldecisions += 1;
+        break;
+      case "constant":
+        break;
+    default:
+        LOG.warn("BullseyeParser unknown probe kind '{}'", kind);
+    }
+  }
+
+  /**
+   * @param event
+   */
+  private static void setTotalCoveredConditions(String event) {
+    switch (event.toLowerCase(Locale.ENGLISH)) {
+      case "full":
+        totalcoveredconditions += 2;
+        break;
+      case "true":
+      case "false":
+        totalcoveredconditions++;
+        break;
+      case "none":
+        // do nothing
+        break;
+      default:
+        // do nothing
+    }
+  }
+
+  /**
+   * @param event
+   */
+  private static void setTotalCoveredDecisions(String event) {
+    switch (event.toLowerCase(Locale.ENGLISH)) {
+      case "full":
+        totalcovereddecisions = 2;
+        break;
+      case "true":        
+      case "false":
         totalcovereddecisions = 1;
-        if ("full".equalsIgnoreCase(event)) {
-          totalcovereddecisions = 2;
-        }
-        if ("none".equalsIgnoreCase(event)) {
-          totalcovereddecisions = 0;
-        }
+        break;
+      default:
```

Comment:
Add also none and a warning like in setTotalCoveredConditions?

Reply:
The "default" and "none" have ```totalcovereddecisions = 0``` which is correct. I think for having "none" explicitly listed might help to understand the mapping. I can add a fall-through for it which I dislike.